Custom serializer (#132)
yiisoft/cache: README.md

- [Redis](https://github.com/yiisoft/cache-redis)
- [Wincache](https://github.com/yiisoft/cache-wincache)

### Data serialization

> The package provide `Yiisoft\Cache\Serializer\SerializerInterface` for data serialization. It's can be useful in database, file

Reviewer: ```suggestion
The package provides `Yiisoft\Cache\Serializer\SerializerInterface` for data serialization. It can be useful in database, file
```

> or redis cache implementations. Out of box you can use `Yiisoft\Cache\Serializer\PhpSerializer` that work via PHP functions
> `serialize()` and `unserialize()`. And you can make own implementation, for example:

Reviewer: ```suggestion
or Redis cache implementations. Out of box, you can use `Yiisoft\Cache\Serializer\PhpSerializer` that works via PHP functions
`serialize()` and `unserialize()`. You can make own implementation, for example:
```

- [Redis](https://github.com/yiisoft/cache-redis)
- [Wincache](https://github.com/yiisoft/cache-wincache)

### Data serialization

> By default, data is serialized using the php functions `serialize/unserialize`. 

Reviewer: ```suggestion
By default, data is serialized using the `serialize()`/`unserialize()` PHP functions. 
```

You can create your own serializer implementing `Yiisoft\Cache\Serializer\SerializerInterface`.

> By default, this is implemented in [Database](https://github.com/yiisoft/cache-db), [File](https://github.com/yiisoft/cache-file) and [Redis](https://github.com/yiisoft/cache-redis).

> > For [APCu](https://github.com/yiisoft/cache-apcu) and [Memcached](https://github.com/yiisoft/cache-memcached) better use php directives `apc.serializer` and `memcached.serializer` accordingly.

Reviewer: ```suggestion
> For [APCu](https://github.com/yiisoft/cache-apcu) and [Memcached](https://github.com/yiisoft/cache-memcached) better use PHP directives `apc.serializer` and `memcached.serializer` accordingly.
```

> By default, data is serialized using the php functions `serialize/unserialize`. 
> You can create your own serializer implementing `Yiisoft\Cache\Serializer\SerializerInterface`.
> 
> > By default, this is implemented in [Database](https://github.com/yiisoft/cache-db), [File](https://github.com/yiisoft/cache-file) and [Redis](https://github.com/yiisoft/cache-redis).
> > For [APCu](https://github.com/yiisoft/cache-apcu) and [Memcached](https://github.com/yiisoft/cache-memcached) better use php directives `apc.serializer` and `memcached.serializer` accordingly.
> 
> For example:
> 
> ```php
> <?php
> 
> use Yiisoft\Cache\Serializer\SerializerInterface;
> 
> final class IgbinarySerializer implements SerializerInterface 
> {
>     public function serialize(mixed $value) : string
>     {
>         return igbinary_serialize($value);
>     }
> 
>     public function unserialize(string $data) : mixed
>     {
>         return igbinary_unserialize($data);
>     }
> }
> ```
> 
> In `di`
> 
> ```php
> 
> use Yiisoft\Cache\Db\DbCache;
> 
> return [
>     DbCache::class => [
>         '__construct()' => [
>              'serializer' => new IgbinarySerializer(),
>         ],
>     ],   
> ];
> ```

Reviewer 2: ```suggestion
The package provide `Yiisoft\Cache\Serializer\SerializerInterface` for data serializition. It's can be useful in database, file
or redis cache implementations. Out of box you can use `Yiisoft\Cache\Serializer\PhpSerializer` that work via PHP functions
`serialize()` and `unserialize()`. And you can make own implementation, for example:

```php
<?php

use Yiisoft\Cache\Serializer\SerializerInterface;

final class IgbinarySerializer implements SerializerInterface 
{
    public function serialize(mixed $value) : string
    {
        return igbinary_serialize($value);
    }

    public function unserialize(string $data) : mixed
    {
        return igbinary_unserialize($data);
    }
}
```
```
